doxygen/contributor_help_pages/require_meta.md

```diff
@@ -0,0 +1,191 @@
+declare specializations of a class. For example, the function below will only work
+on types derived from [`Eigen::DenseBase`](https://eigen.tuxfamily.org/dox/classEigen_1_1DenseBase.html) with only 1 row or column at compile time (aka types such as `Eigen::Matrix<double, -1, 1>` or `Eigen::Matrix<double, 1, -1>`).
+
+```cpp
+template <typename EigVec, require_eigen_vector_t<EigVec>* = nullptr>
+auto my_func(const EigVec& x) {
+  // ...
+}
+```
+
+The `require` is included in the template as a pointer [non-type template parameter](https://en.cppreference.com/w/cpp/language/template_parameters#Non-type_template_parameter) with a default value of `nullptr`. This might look a bit odd, but it uses the fact that any non-type template parameter of type `void*` with a default of `nullptr` will be ignored by the compiler. Under the hood, if any of the `requires` are successful they will return back a type `void`. So when we pass a type that the `require` accepts we get back a `void* = nullptr`, which is safely ignored by the compiler. In the case that the type does not satisfy the `require` then the function is removed from the set of possible functions the caller could use via SFINAE. With this scheme we end up having a very nice pattern for writing generic templates for functions while also being able to restrict the set of types that a function can be used for.
+
+For overloading classes and structs with this scheme we create an initial forward definition with a `void` non-type template parameter. Then the class overloads use the requires in place of the non-type template parameter.
+
+```cpp
+template <typename T, typename = void>
+class a_class;
+
+/**
+ * Overload for standard vectors with \ref stan::scalar_type
+ *  of \ref stan::math::var
+ */
+template <typename T>
+class a_class<T, require_std_vector_st<is_var, T>> {
+  // ...
+};
+```
+
+As noted in the docs above, `a_class` has an overload specifically for standard vectors with @ref stan::scalar_type 's of @ref stan::math::var .
+
+Below covers the general themes for all of the [`requires`](@ref require_meta) found
+in the Stan math library. Any `*` should be thought of as a wildcard where a type
+traits name is put in its place.
+
+- `requires_*_t`: A template parameter `T` must satisfy the require in order for
+the overload to be available.
+
+```cpp
+ // Works for stan::math::var types
+ template <typename T, require_var_t<T>* = nullptr>
+ auto add(T&& x, T&& y) { return x + y; }
+```
+
+- `require_not_*_t` : A template parameter `T` must *not* satisfy the require in order for
+the overload to be availabe.
+
+*NOTE:* The `not` version of requires should be used sparingly. Often `requires` are used to specify, conceptually, what types a function should accept. Defining a function by the types it cannot accept can make understanding what goes into a function more difficult and error prone.
+
+```cpp
+ // Works for non standard vector types
```

Comment:
What's a non standard vector type?  Shouldn't this work for any x and y other than std::vector?  Are there other "standard vector types"?

Reply:
Fixed the wording here, should just say that this function works with any type this a not a `std::vector`